artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/AMQPBrokerConnection.java

```diff
@@ -490,12 +495,18 @@ private void connectSender(Queue queue,
             Source source = new Source();
             source.setAddress(queue.getAddress().toString());
             sender.setSource(source);
+            HashMap<Symbol, Object> mapProperties = new HashMap<>(1);
```

Comment:
@gemmellr I'm only using a single element here. as I read the code, when you call putVal on hashMap, it will resize for 1 element, which is my intention.

Reply:
Try debugging it. Unless its changed, it will resize itself again immediately during/after insertion due to exceeding the load-factor.